src/emulator/storage/apis/gcloud.ts

```diff
@@ -39,38 +40,54 @@ export function createCloudEndpoints(emulator: StorageEmulator): Router {
 
   gcloudStorageAPI.get(
     ["/b/:bucketId/o/:objectId", "/download/storage/v1/b/:bucketId/o/:objectId"],
-    (req, res) => {
-      const md = storageLayer.getMetadata(req.params.bucketId, req.params.objectId);
-
-      if (!md) {
-        res.sendStatus(404);
-        return;
+    async (req, res) => {
+      let getObjectResponse: GetObjectResponse;
+      try {
+        getObjectResponse = await storageLayer.handleGetObject(
+          {
+            bucketId: req.params.bucketId,
+            decodedObjectId: req.params.objectId,
+          },
+          /* skipAuth = */ true
+        );
+      } catch (err) {
+        if (err instanceof NotFoundError) {
+          return res.sendStatus(404);
+        }
+        if (err instanceof ForbiddenError) {
+          throw new Error("Request failed unexpectedly due to Firebase Rules.");
```

Comment:
Why are we not returning a `403` here?

Reply:
Same comment as above, this is to signal a critical error in the emulator, as the current assumption is this endpoint should never fail with a 403